src/main/java/org/pac4j/vertx/handler/impl/RequiresAuthenticationHandler.java

```diff
@@ -57,15 +39,16 @@
     private static final Logger LOG = LoggerFactory.getLogger(RequiresAuthenticationHandler.class);
 
     protected final Config config;
-    protected final String clientName;
+
+    protected final String clientNames;
     protected final String authorizerName;
     protected final String matcherName;
+    protected final boolean multiProfile;
     protected final Vertx vertx;
 
-    protected HttpActionAdapter httpActionAdapter = new DefaultHttpActionAdapter();
-    protected ClientFinder clientFinder = new DefaultClientFinder();
-    protected AuthorizationChecker authorizationChecker = new DefaultAuthorizationChecker();
-    protected MatchingChecker matchingChecker = new DefaultMatchingChecker();
+    protected HttpActionAdapter<Void, VertxWebContext> httpActionAdapter = new DefaultHttpActionAdapter();
+
+    private SecurityLogic<Void, VertxWebContext> securityLogic = new VertxSecurityLogic();
 
     public RequiresAuthenticationHandler(final Vertx vertx, final Config config, final Pac4jAuthProvider authProvider,
```

Comment:
This should be renamed as `SecurityHandler` with consistency with other implementations.